tsp/tsp_client.py

```diff
@@ -252,6 +253,52 @@ def fetch_timegraph_tree(self, exp_uuid, output_id, parameters=None):
             print("failed to get tree: {0}".format(response.status_code))
             return TspClientResponse(None, response.status_code, response.text)
 
+    def fetch_xy_tree(self, exp_uuid, output_id, parameters=None):
+        '''
+        Fetch XY tree, Model extends Entry
+        :param exp_uuid: Experiment UUID
+        :param output_id: Output ID
+        :param parameters: Query object
+        :returns: :class:  `TspClientResponse <GenericResponse>` object XY entry response with entries and headers
+        :rtype: TspClientResponse
+        '''
+        api_url = '{0}experiments/{1}/outputs/XY/{2}/tree'.format(
+            self.base_url, exp_uuid, output_id)
+
+        params = parameters
+        if (parameters is None):
+            requested_times = [0, 1]
```

Comment:
> The TSP states for all XY_TIME_TREE, DATA_TREE and TIME_GRAPH:
> 
> "Query parameters to fetch the XY tree. When 'requested_times' is absent the tree for the full range is returned. When present it specifies a range as [start, end]".

Above, in the TSP spec, the term `absent` should be replaced with `empty`. Because the absent case is not properly supported by the server without 500 errors. As for the empty case, it works better with [TimeQueryFilter: Fix 500 if empty fTimesRequested](https://git.eclipse.org/r/c/tracecompass/org.eclipse.tracecompass/+/185164). You also mentioned that `empty` expectation somehow down below indeed.

> I suggest to support full range (trace start to trace end) only, and then support specific range.
> 
> Please note, that it seems to me that the Trace Compass server doesn't fully comply with this.
> I think, all TIME_GRAPHs return the all tree nodes regardless of the time range. Also, it looks like that the requested_times have to be there regardless of the content.
> Many XY_TIME_TREE on the other hand require a valid time range to get rows that are active in that time range. For example, org.eclipse.tracecompass.analysis.os.linux.core.cpuusage.CpuUsageDataProvider for LTTng kernel traces.
> For DATA_TREE, it's probably similar, but we need to verify.
> 
> Items to backlog:
> 
> * support time ranges for queries
> * make sure that Trace Compass server complies with TSP

Can you please back-log these items, given your more extensive knowledge of the whole situation or context?

Reply:
> > The TSP states for all XY_TIME_TREE, DATA_TREE and TIME_GRAPH:
> > "Query parameters to fetch the XY tree. When 'requested_times' is absent the tree for the full range is returned. When present it specifies a range as [start, end]".
> 
> Above, in the TSP spec, the term `absent` should be replaced with `empty`. Because the absent case is not properly supported by the server without 500 errors. As for the empty case, it works better with [TimeQueryFilter: Fix 500 if empty fTimesRequested](https://git.eclipse.org/r/c/tracecompass/org.eclipse.tracecompass/+/185164). You also mentioned that `empty` expectation somehow down below indeed.

I think this should be fixed in the DataProviderService instead, as described here: https://bugs.eclipse.org/bugs/show_bug.cgi?id=575950.

Reply:
> 
> > I suggest to support full range (trace start to trace end) only, and then support specific range.
> > Please note, that it seems to me that the Trace Compass server doesn't fully comply with this.
> > I think, all TIME_GRAPHs return the all tree nodes regardless of the time range. Also, it looks like that the requested_times have to be there regardless of the content.
> > Many XY_TIME_TREE on the other hand require a valid time range to get rows that are active in that time range. For example, org.eclipse.tracecompass.analysis.os.linux.core.cpuusage.CpuUsageDataProvider for LTTng kernel traces.
> > For DATA_TREE, it's probably similar, but we need to verify.
> > Items to backlog:
> > 
> > * support time ranges for queries
> > * make sure that Trace Compass server complies with TSP
> 
> Can you please back-log these items, given your more extensive knowledge of the whole situation or context?

I will do that.

Reply:
Here are the items:

- #36
- https://bugs.eclipse.org/bugs/show_bug.cgi?id=575971 and https://bugs.eclipse.org/bugs/show_bug.cgi?id=575950